### DISCUSSION TITLE
Etherscan Invalid API Key error?

### DISCUSSION THREAD

**Discussion by jpdnft** [NomicFoundation/hardhat#4787] [Troubleshooting] [answered]
Getting this today using npx hardhat verify on sepolia

An unexpected error occurred during the verification process.
Please report this issue to the Hardhat team.
Error Details: Invalid API Key

According to Etherscan, the API is fine, and so is my API key. So, I'm not sure what's going on. Ideas?

**Comment by fvictorio** [comment]
Can you run `curl 'https://api.etherscan.io/api?module=account&action=balance&address=0x0000000000000000000000000000000000000000&tag=latest&apikey=<YOUR-API-KEY>` in your terminal (using your API key) to triple-check that the API key is indeed working fine?

**Comment by jpdnft** [comment, marked as answer]
Yeah, it was some weird Etherscan fluke... problem went away on its own after several hours of me wondering what was up. API key was valid/working fine.  :-)

**Comment by hgky95** [comment]
For future reference, if you're experiencing problems with the Invalid API Key, check their status page at [Etherscan Status](https://etherscan.freshstatus.io/ ) to see if there are any ongoing issues. 
The platform might be in maintenance mode, which could explain any service disruptions you're encountering.
